Fix that ducking revolver issue

toberge opened the pull request:
pls merge the other one first....

or just review this as both of them maybe

Outlining the issue in Norwegian:
- skyt nest siste skudd (her er ammo > 1)
- vent lenge nok til at animasjonen må være ferdig (mens anim kjører er ammo fortsatt > 1)
- prøv å skyte siste skudd (her er ammo == 1 og også <= 1)
--> da får du ikke skutt

This ammo <= 1 check did fix the animation issue we had with you being able to trigger fire animation just as the body started reloading, but removing this not-fix meant I had to fix it otherwise. Adjusted animation speed to actually correspond to the animation length, and gave the laser barrel an actual reduction in fire rate, and now things seem to work smoothly.

Closes #970

toberge committed:
Fix revolver + lazur/coil not firing

Outlining the issue in Norwegian:
- skyt nest siste skudd (her er ammo > 1)
- vent lenge nok til at animasjonen må være ferdig (mens anim kjører er ammo fortsatt > 1)
- prøv å skyte siste skudd (her er ammo == 1 og også <= 1)
--> da får du ikke skutt

This ammo <= 1 check did fix the animation issue we had with you being able to trigger fire animation just as the body started reloading, but removing this not-fix meant I had to fix it otherwise. Adjusted animation speed to actually correspond to the animation length, and gave the laser barrel an actual reduction in fire rate, and now things seem to work smoothly.

toberge committed:
Buff brick barrel

It is really underwhelming currently 😔

toberge commented:
may have some funky results with ddr + lazur

toberge committed:
Increment version number to 0.5.3

Fueredoriku commented on Assets/Scripts/Augment/GunController.cs:
```diff
@@ -81,6 +82,8 @@ public void SetPlayerInput(InputManager input)
     private AugmentAnimator barrelAnimator;
 
+    // TODO denne er faktisk ikke i bruk nå siden den ikke blir satt til true lengre.
+    //      -> men hvis du setter den når vi skyter så brekker du spillet
```
Maybe remove it in that case?

toberge replied:
added more to the comment. ideally we'd have a replacement

Fueredoriku replied:
Yeah this seems like something we need to get to the bottom of at some point. My hunch is that there is some animation event firing somewhere that causes the crash

toberge replied:
Yeah, I'll actually remove this source of confusion I suppose

Fueredoriku commented on Assets/Scripts/Augment/GunController.cs:
```diff
@@ -254,6 +277,7 @@ private void FireGun()
 
         try
         {
+            isFiring = true;
```
I suspect something in this block breaks lazur when continously firing with an automatic firemode

toberge replied:
wait shit did I forget to remove that line :skull:

toberge replied:
removed, should work now

Fueredoriku reviewed (changes requested):
Very nice to finally fix the revolver!! 😮

Lazur seems really broken now though, using an automatic firemode body like seeker will break the lazur if holding fire contiously. I think this must be fixed before merge

toberge committed:
Fix trash uplink cleanup nullreference

We really shouldn't have to set the pools to null explicitly right?

Fueredoriku reviewed (approved):
It works! Lets goooo!

toberge committed:
Remove isFiring entirely as it does not work

Adding isFiring = true at the appropriate place causes bugs, so this
approach does not work when paired with all our current jank :/
